src/query/storage/promremote/types.go

```diff
@@ -34,12 +39,61 @@ type Options struct {
 	endpoints   []EndpointOptions
 	httpOptions xhttp.HTTPClientOptions
 	scope       tally.Scope
+	logger      *zap.Logger
+}
+
+// Namespaces returns M3 namespaces from endpoint opts.
+func (o Options) Namespaces() m3.ClusterNamespaces {
+	namespaces := make(m3.ClusterNamespaces, len(o.endpoints))
+	for i, endpoint := range o.endpoints {
+		metricType := storagemetadata.AggregatedMetricsType
+		if endpoint.resolution == 0 {
+			metricType = storagemetadata.UnaggregatedMetricsType
+		}
+		namespaceOptions := m3.NewClusterNamespaceOptions(
+			storagemetadata.Attributes{
+				MetricsType: metricType,
+				Resolution:  endpoint.resolution,
+				Retention:   endpoint.retention,
+			},
+			&m3.ClusterNamespaceDownsampleOptions{
+				All: endpoint.downsampleAll,
```

Comment:
I think it does not really matter https://github.com/m3db/m3/blob/b1eb46efa3f4e123a898c099ca2f28958197688f/src/query/storage/m3/cluster.go#L121

Not sure if I like duplicating this logic here. From config perspective if you set storage policy - it means namespace is aggregated. If you don't set storage policy - it will be unaggregated. You will not be able to set downsample then and here it will be false.